Controller/AdminResettingController.php

```diff
@@ -11,143 +11,179 @@
+        if ($this->get('security.authorization_checker')->isGranted('IS_AUTHENTICATED_FULLY')) {
+            return new RedirectResponse($this->get('router')->generate('sonata_admin_dashboard'));
         }
 
-        $user = $this->container->get('fos_user.user_manager')->findUserByConfirmationToken($token);
+        /** @var $formFactory \FOS\UserBundle\Form\Factory\FactoryInterface */
+        $formFactory = $this->get('fos_user.resetting.form.factory');
+        /** @var $userManager \FOS\UserBundle\Model\UserManagerInterface */
+        $userManager = $this->get('fos_user.user_manager');
+        /** @var $loginManager \FOS\UserBundle\Security\LoginManagerInterface */
+        $loginManager = $this->get('fos_user.security.login_manager');
+
+        $user = $userManager->findUserByConfirmationToken($token);
+
+        $ttl = $this->container->getParameter('fos_user.resetting.retry_ttl');
+        $firewallName = $this->container->getParameter('fos_user.firewall_name');
 
         if (null === $user) {
             throw new NotFoundHttpException(sprintf('The user with "confirmation token" does not exist for value "%s"', $token));
         }
 
-        if (!$user->isPasswordRequestNonExpired($this->container->getParameter('fos_user.resetting.token_ttl'))) {
-            return new RedirectResponse($this->container->get('router')->generate('sonata_user_admin_resetting_request'));
+        if (!$user->isPasswordRequestNonExpired($ttl)) {
+            new RedirectResponse($this->generateUrl('sonata_user_admin_resetting_request'));
         }
 
-        $form = $this->container->get('fos_user.resetting.form');
-        $formHandler = $this->container->get('fos_user.resetting.form.handler');
-        $process = $formHandler->process($user);
+        $form = $formFactory->createForm();
+        $form->setData($user);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $user->setConfirmationToken(null);
+            $user->setPasswordRequestedAt(null);
+            $user->setEnabled(true);
+
+            $message = $this->get('translator')->trans('resetting.flash.success', array(), 'FOSUserBundle');
+            $this->addFlash('success', $message);
+            $response = new RedirectResponse($this->generateUrl('sonata_admin_dashboard'));
 
-        if ($process) {
-            $this->setFlash('fos_user_success', 'resetting.flash.success');
-            $response = new RedirectResponse($this->container->get('router')->generate('sonata_admin_dashboard'));
-            $this->authenticateUser($user, $response);
+            try {
+                $loginManager->logInUser($firewallName, $user, $response);
+                $user->setLastLogin(new \DateTime());
+            } catch (AccountStatusException $ex) {
+                // We simply do not authenticate users which do not pass the user
+                // checker (not enabled, expired, etc.).
```

Comment:
@greg0ire I am not sure about this one in terms of new dependencies. Should I add monolog to require section? or just check if container has "logger" before calling it out?